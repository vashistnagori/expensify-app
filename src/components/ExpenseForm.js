import React from 'react';
import moment from 'moment';
import {SingleDatePicker} from 'react-dates';
//import 'react-dates/lib/css/_datepicker.css'; now in app.js file

export default class ExpenseForm extends React.Component{
state={
                description: this.props.expense ?  this.props.expense.description : "",
                note: this.props.expense ?  this.props.expense.note : "",
                amount: this.props.expense ? ( this.props.expense.amount / 100).toString() : "",
                createdAt: this.props.expense ?  moment(this.props.expense.createdAt) : moment(),
                calanderFocused:false,
                error:"",
        };
 decriptionChange=(e)=>{
     const description= e.target.value;
     this.setState(()=>({description}));
 };
 noteChange=(e)=>{
    const note= e.target.value;
    this.setState(()=>({note}));
};
amountChange=(e)=>{
    const amount= e.target.value;
    if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
        this.setState(()=>({amount}));
    }
};
dateChange=(createdAt)=>{
    if(createdAt){
        this.setState(()=>({createdAt}));
    }
}
forcusChange=({focused})=>{
this.setState(()=>({calanderFocused:focused}));
};
formSubmit=(e)=>{
    e.preventDefault();
    if(!this.state.description || !this.state.amount){
        this.setState(()=>({error:"please provide decription and amount"}));
    }
    else{
        this.setState(()=>({error:""}));
        this.props.onSubmit({
            description: this.state.description,
            amount:this.state.amount*100,
            createdAt:this.state.createdAt.valueOf(),
            note: this.state.note
        });
    }
};


    render(){
        return(<div>
           {this.state.error && <p>{this.state.error}</p> } 
            <form onSubmit={this.formSubmit}>
            <input 
            type='text'
            placeholder='description'
            value={this.state.description}
            onChange={this.decriptionChange}
            autoFocus
            />
            <input 
            type='number'
            placeholder='amount'
            value={this.state.amount}
            onChange={this.amountChange}
            />
            <SingleDatePicker
            date={this.state.createdAt} 
            onDateChange={this.dateChange} 
            focused={this.state.calanderFocused}
            onFocusChange={this.forcusChange} 
            numberOfMonths={1}
            isOutsideRange={()=>false}
            />
            <textarea
            placeholder="add a note here"
            value={this.state.note}
            onChange={this.noteChange}>
            </textarea>
            <button>Add Expense</button>
            </form>
            </div>);
    }
}