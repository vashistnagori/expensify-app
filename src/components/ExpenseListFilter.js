import React from 'react';
import {connect} from 'react-redux';
import * as filters_Actions from "../actions/filters";
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilter extends React.Component{
    state={
        calanderFocused:null,

    };
    onDateChange=(dates)=>{
        this.props.dispatch_setStartDate(dates.startDate);
        this.props.dispatch_setEndDate(dates.endDate);
    };
    onFocusChange=(f)=>{
            this.setState(()=>({
                calanderFocused:f
            }))
    };
    onTextChange=(e)=>{
        this.props.dispatch_setTextFilter(e.target.value);
    };
    onSortChange=(e)=>{
        if(e.target.value=="Date"){
         this.props.dispatch_sortByDate();
        }
        else if(e.target.value=="Amount"){
         this.props.dispatch_sortByAmount();
     }
     };
render(){
    return(<div>
        <input type="text" 
        value={this.props.filters.text} 
        onChange={this.onTextChange}/>
        <select
         value={this.props.filters.SortBy} 
         onChange={this.onSortChange}>
        <option>Date</option>
        <option>Amount</option>
        </select>
        <DateRangePicker
        startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
        endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
        onDatesChange={this.onDateChange} // PropTypes.func.isRequired,
        focusedInput={this.state.calanderFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
        onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
        numberOfMonths={1}
        isOutsideRange={()=>false}
        showClearDates={true}
        />
        </div>);
};
};
    const stateMapToProps=(state)=>{
        return {
            filters:state.filter
        }
    };

    const DispatchToProps=(dispatch)=>{
        return {
            dispatch_setTextFilter:(text)=>dispatch(filters_Actions.setTextFilter(text)),
            dispatch_sortByDate:()=>dispatch(filters_Actions.sortByDate()),
            dispatch_sortByAmount:()=>dispatch(filters_Actions.sortByAmount()),
            dispatch_setStartDate:(startdate)=>dispatch(filters_Actions.setStartDate(startdate)),
            dispatch_setEndDate:(enddate)=>dispatch(filters_Actions.setEndDate(enddate))
        }
    };
    export default connect(stateMapToProps,DispatchToProps )(ExpenseListFilter);



