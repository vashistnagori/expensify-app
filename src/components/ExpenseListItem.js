import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem= ({id, description, amount, createdAt})=>{
    const to="/edit/"+id;
return (<Link className="list-item"  to={to}>
    <div>
    <h3 className="list-item__title">{description}</h3>
    <span className="list-item__sub- title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
    </div>
    <h3 className="list-item__data">{numeral(amount / 100).format('$0,0.00')}</h3>
   </Link>)};

export default ExpenseListItem;
