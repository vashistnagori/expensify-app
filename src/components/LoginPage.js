import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

 export const LoginPage=(props)=>{
    return (<div className="box-layout">
            <div className="box-layout__box"> 
            <h1 className="box-layout__title">Expensify</h1>
            <p>Calculate your expenses. App by Vashist Nagori</p>
            <button onClick={props.startLogin} className="button">Login with Google</button>
            </div>
        </div>);
};


const mapDispatchToprops=(dispatch)=>({
startLogin:(dispatch)=>{startLogin()(dispatch)}
});

export default connect(undefined,mapDispatchToprops)(LoginPage);

