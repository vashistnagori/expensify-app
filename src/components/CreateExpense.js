import React from "react";
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startAddExpense} from '../actions/expense';

export class CreateExpense extends React.Component{
  onSubmit=(expense)=>{
    this.props.dispatch_startAddExpense(expense);
    this.props.history.push("/");
  }
  render(){
    return( <div>
      <div className="page-header ">
      <div className="content-container">
      <h1 className="page-header__title">Add Expense</h1>
      </div>
      </div>
      <div className="content-container"> <ExpenseForm onSubmit={this.onSubmit}/></div> 
      </div>);
  }
}

    const mapDispatchToProps=(dispatch)=>({
        dispatch_startAddExpense:(expense)=>dispatch(startAddExpense(expense))
    });

export default connect(undefined, mapDispatchToProps)(CreateExpense);
