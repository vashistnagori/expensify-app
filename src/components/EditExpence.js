import React from "react";
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense, startRemoveExpense, startEditExpense} from '../actions/expense';

// export const EditExpence=(props)=>{
//    return(
//     <div>
//     <ExpenseForm 
//      expense={props.expense}  
//      onSubmit={(expense)=>{
//         props.dispatch_editExpense(props.expense.id, expense);
//         props.history.push("/");
//      }}/>
//      <button onClick={(e)=>{
//         props.dispatch_removeExpense({id:props.expense.id});
//         props.history.push("/");
//        }}>remove</button>
//     </div>
//     );
//    }

export class EditExpence extends React.Component{
   onSubmitfun=(expense)=>{
      this.props.dispatch_editExpense(this.props.expense.id, expense);
      this.props.history.push("/");
   }
   onDelFun=(e)=>{
      this.props.dispatch_removeExpense({id:this.props.expense.id});
      this.props.history.push("/");
     }

   render(){
      return(
             <div>
             <ExpenseForm 
              expense={this.props.expense}  
              onSubmit={this.onSubmitfun}/>
              <button onClick={this.onDelFun}>remove</button>
             </div>
             );
   }
}

const mapStateToProps=(state, props)=>{
   const expense= state.expenses.find((e)=>{
      return e.id === props.match.params.id;
   });
   if(expense){ return {expense}; } else{ return {};  }
}

const mapDispatchToProps=(dispatch)=>{
return{
   dispatch_removeExpense:({id})=>startRemoveExpense({id})(dispatch),
   dispatch_editExpense:(id, expense)=> startEditExpense(id, expense)(dispatch)
}
}


export default connect(mapStateToProps, mapDispatchToProps)(EditExpence);