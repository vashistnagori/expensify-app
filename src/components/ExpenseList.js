import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import {filterExpense} from '../selectors/expense';


export const ExpenseList =(props)=>(<div className="content-container">
<div className="list-header">
<div className="show-for-mobile">Expenses</div> 
<div className="show-for-desktop">Expense</div>
<div className="show-for-desktop">Amount</div>
 </div> <div className="list-body ">
{
    props.expenses.length !== 0 ? (props.expenses.map((c)=>{
                                        return (<div>
                                        <ExpenseListItem key={"list"+c.id} {...c} />
                                        </div>);
                                        })
    ) : (<div className="list-item--message ">
        <span>No Expenses</span></div>)
}
    </div></div>);


const mapStateToProps=(state)=>{
    return {
        expenses: filterExpense(state.expenses,state.filter)
    }
}

export default connect(mapStateToProps)(ExpenseList);
