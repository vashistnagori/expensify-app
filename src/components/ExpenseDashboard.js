import React from "react";
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';
import {ExpenseSumary} from './ExpenseSumary';
import {connect} from 'react-redux';
import {expenseTotal} from '../selectors/expense-total';


const ExpenseDashboardMaking=(props)=>{
const total= expenseTotal(props.expenses);
   return(<div>
    <p>this is homepage</p>
    <ExpenseListFilter />
    <ExpenseSumary total={total} count={props.expenses.length}/>
    <ExpenseList />
    </div>);
}

const mapStateToProps =(state)=>({
    expenses:state.expenses
});

const ExpenseDashboard= connect(mapStateToProps)(ExpenseDashboardMaking);


export default ExpenseDashboard;