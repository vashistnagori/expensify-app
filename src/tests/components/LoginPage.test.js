import React from 'react';
import {shallow } from 'enzyme';
import {LoginPage} from '../../components/LoginPage';

test("Testing login Page component",()=>{
const wrapper= shallow(<LoginPage startLogin={()=>{}}/>);
expect(wrapper).toMatchSnapshot();
});

test("Testing login function on login Page ",()=>{
    const skyLoginFunction=jest.fn();
    const wrapper= shallow(<LoginPage startLogin={skyLoginFunction}/>);
    wrapper.find('button').simulate("click");
    expect(skyLoginFunction).toHaveBeenCalledWith();
    });