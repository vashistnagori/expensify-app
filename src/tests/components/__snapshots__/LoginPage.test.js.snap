// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login Page component 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      Calculate your expenses. App by Vashist Nagori
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
